### ISSUE TITLE
testing DOM performance.

### ISSUE THREAD

**Issue by trusktr** [lukejacksonn/perflink#35]
I wanted to test DOM performance, [for example](https://perf.link/#eyJpZCI6InYwbWM1N3VrdTRnIiwidGl0bGUiOiJbRE9NIHZpYSBzdHJpbmddIHZzIFthcHBlbmQoKSByZXR1cm5pbmcgdGhpc10iLCJiZWZvcmUiOiJjb25zdCBkYXRhID0ge1xuICBuYW1lOiAnSm9lJyxcbiAgYWdlOiAzNCxcbn1cblxuY29uc3Qgb3JpZ2luYWxBcHBlbmQgPSBFbGVtZW50LnByb3RvdHlwZS5hcHBlbmRcbkVsZW1lbnQucHJvdG90eXBlLmFwcGVuZCA9IGZ1bmN0aW9uIGFwcGVuZCguLi5hcmdzKSB7XG4gIG9yaWdpbmFsQXBwZW5kLmFwcGx5KHRoaXMsIGFyZ3MpXG4gIHJldHVybiB0aGlzXG59XG5cbmZ1bmN0aW9uIHJlbmRlcldpdGhTdHJpbmcoZGF0YSkge1xuICBkb2N1bWVudC5ib2R5Lmluc2VydEFkamFjZW50SFRNTCgnYmVmb3JlZW5kJywgYFxuICAgIDxkaXY%2BPHNwYW4%2BbmFtZTogJHtkYXRhLm5hbWV9PC9zcGFuPiAtIDxzcGFuPmFnZTogJHtkYXRhLmFnZX08L3NwYW4%2BPC9kaXY%2BXG4gIGApIFxufVxuXG5mdW5jdGlvbiByZW5kZXJXaXRoQXBwZW5kKGRhdGEpIHtcbiAgZG9jdW1lbnQuYm9keS5hcHBlbmQoXG4gICAgZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnZGl2JykuYXBwZW5kKFxuICAgICAgZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnc3BhbicpLmFwcGVuZChcbiAgICAgICAgJ25hbWU6ICcrZGF0YS5uYW1lXG4gICAgICApLFxuICAgICAgJyAtICcsXG4gICAgICBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdzcGFuJykuYXBwZW5kKFxuICAgICAgICAnYWdlOiAnK2RhdGEubmFtZVxuICAgICAgKSxcbiAgICApLFxuICApXG59IiwidGVzdHMiOlt7Im5hbWUiOiJGaW5kIGl0ZW0gMTAwIiwiY29kZSI6InJlbmRlcldpdGhTdHJpbmcoZGF0YSkiLCJvcHMiOjB9LHsibmFtZSI6IkZpbmQgaXRlbSAyMDAiLCJjb2RlIjoicmVuZGVyV2l0aEFwcGVuZChkYXRhKSIsIm9wcyI6MH1dLCJ1cGRhdGVkIjoiMjAyMC0wOC0yN1QwNzo0NToyMi4zOTdaIn0%3D).

But `window` or `Element` are not defined. I'm guessing it's running the code in a worker. Would be sweet if we had an option to run in an iframe instead.

**Comment by lukejacksonn** [comment]
You are correct about the workers.. this was quite the oversight when going from V1 fo V2 😅

Emulating a thread via an iframe seems like a pretty good middle ground between running on the main thread (V1) and running in workers (V2).

There really isn't that much code in this repo (albeit perhaps not all that well organised) and I am not that precious about how things are wired up. So if you would like to take a stab at this then I'd be happy to review a PR.

Are you imagining an iframe/worker toggle switch somewhere?

**Comment by trusktr** [comment]
Yeah, perhaps a toggle would do it. let me see if I can circle back to this.

**Comment by lukejacksonn** [comment]
Awesome! You could probably get most of the code from old (pre V2) commits in the repo.
